publisher/publisher.go

```diff
@@ -111,7 +112,16 @@ func (pub *publisher) Publish(contentType string, content []byte, config map[str
 		tags := map[string]string{}
 
 		// at least one tag is required by KairosDB
-		tags["hostname"] = metric.Source()
+		if hostname, ok := metric.Tags()["hostname"]; ok {
+			tags["hostname"] = hostname
+		} else {
+			hostname, err := os.Hostname()
```

Comment:
Could have this be `hostname, err:= metric.Tags()[core.STD_TAG_PLUGIN_RUNNING_ON]` since this should be the os.hostname.